web/packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -640,3 +640,24 @@ export function makeApp(source: tsh.App): App {
 
   return { ...source, addrWithProtocol };
 }
+
+/** Returns a URL that can be used to open the app in a browser. */
+export function getWebAppLaunchUrl(
+  app: tsh.App,
+  clustersService: ClustersService
+): string {
+  const { fqdn, publicAddr } = app;
+
+  const rootCluster = clustersService.findCluster(
+    routing.ensureRootClusterUri(app.uri)
+  );
+  const cluster = clustersService.findClusterByResource(app.uri);
+
+  const canCreateUrl =
+    rootCluster.proxyHost && fqdn && cluster?.name && publicAddr;
+
+  if (!canCreateUrl) {
+    return '';
+  }
+  return `https://${rootCluster.proxyHost}/web/launch/${fqdn}/${cluster.name}/${publicAddr}`;
```

Comment:
I checked it in my local cluster where the name of the cluster is different than the proxy host of it and this launch URL works fine. Both when I log in directly to the cluster and when it acts as a leaf cluster.